terraform/main.tf

```diff
@@ -0,0 +1,61 @@
+# Enable Google Cloud APIs
+module "enable_google_apis" {
+  source                      = "terraform-google-modules/project-factory/google//modules/project_services"
+  version                     = "~> 13.0"
+
+  project_id                  = var.gcp_project_id
+  disable_services_on_destroy = false
+
+  activate_apis               = var.apis
+}
+
+# Create GKE Cluster
+resource "google_container_cluster" "cluster" {
+  name               = var.name
+  location           = var.region
+  ip_allocation_policy {
+  }
+
+  # Enabling Autopilot for this cluster
+  enable_autopilot = true
+
+  depends_on = [
+    module.enable_google_apis
+  ]
+}
+
+# Get Credentials for Cluster
```

Comment:
(nit) "the cluster"

also I would be consistent with only capitalizing the first letter of the comment and rest lowercase unless it's a special term of abbreviation (e.g. GKE)